src/parser/jobs/mnk/modules/RiddleOfEarth.tsx

```diff
@@ -0,0 +1,84 @@
+import {t} from '@lingui/macro'
+import {Plural, Trans} from '@lingui/react'
+import React from 'react'
+
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+
+import {BuffEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+
+class Riddle {
+	active: boolean = false
+	clean: boolean = false
+	start: number
+	end?: number
+
+	constructor(start: number) {
+		this.start = start
+	}
+}
+
+export default class RiddleOfEarth extends Module {
+	static handle = 'riddleofearth'
+
+	@dependency private suggestions!: Suggestions
+
+	private history: Riddle[] = []
+	private riddle?: Riddle
+
+	protected init(): void {
+		this.addHook('applybuff', {to: 'player', abilityId: STATUSES.RIDDLE_OF_EARTH.id}, this.onGain)
+		this.addHook('applybuff', {to: 'player', abilityId: STATUSES.EARTHS_REPLY.id}, this.onReply)
+		this.addHook('removebuff', {to: 'player', abilityId: STATUSES.RIDDLE_OF_EARTH.id}, this.onDrop)
+
+		this.addHook('complete', this.onComplete)
+	}
+
+	private onGain(event: BuffEvent): void {
+		this.riddle = new Riddle(event.timestamp)
+		this.riddle.active = true
```

Comment:
It doesn't look like `active` is really doing much. You're setting it to `true` in the only place `Riddle` is constructed, and it isn't set to `false` until it's in the history. Worth nuking?

Reply:
It's mainly there to make sure we only push an open window at the end, instead of accidentally duping I guess. Maybe I'm being overly cautious?